tiatoolbox/wsicore/wsireader.py

```diff
@@ -362,56 +365,57 @@ def _find_read_params_at_resolution(
         # Do we need sanity check for input form ?
         requested_location = np.array(location)
         requested_size = np.array(size)
-        lv0_to_read_lv_scale_factor = 1 / info.level_downsamples[read_level]
+        baseline_to_read_level_scale_factor = 1 / info.level_downsamples[read_level]
 
-        lv0_to_requested_lv_scale_factor = lv0_to_read_lv_scale_factor
-        lv0_to_requested_lv_scale_factor *= read_lv_to_requested_scale_factor
+        baseline_to_resolution_scale_factor = baseline_to_read_level_scale_factor
+        baseline_to_resolution_scale_factor *= read_level_to_resolution_scale_factor
 
-        size_at_lv0 = requested_size / lv0_to_requested_lv_scale_factor
-        location_at_lv0 = requested_location / lv0_to_requested_lv_scale_factor
-        size_at_read_lv = requested_size / read_lv_to_requested_scale_factor
-        location_at_read_lv = requested_location / read_lv_to_requested_scale_factor
+        size_at_baseline = requested_size / baseline_to_resolution_scale_factor
+        location_at_baseline = np.array(requested_location, dtype=np.float32)
+        location_at_baseline /= baseline_to_resolution_scale_factor
```

Comment:
Same thing here regarding `/=` and `*=` but looks like you are making a copy above with `np.array`. This could be more clear by uinsg a regular division or `.copy` as `requested_location` is already an array. Fore example:

```python
location_at_baseline = np.array(requested_location, dtype=np.float32)
location_at_baseline /= baseline_to_resolution_scale_factor
```

could simply be

```python
location_at_baseline = requested_location  / baseline_to_resolution_scale_factor
```